addons/base_vat/models/account_fiscal_position.py

```diff
@@ -13,12 +13,29 @@ def _validate_foreign_vat(self):
             if not record.foreign_vat:
                 continue
 
-            checked_country_code = self.env['res.partner']._run_vat_test(record.foreign_vat, record.country_id)
+            if record.country_group_id:
+                # Checks the foreign vat is a VAT Number linked to a country of the country group
+                foreign_vat_country_id = next((country_id for country_id in self.country_group_id.country_ids if country_id.code.upper() == self.foreign_vat[:2]), False)
```

Comment:
If you call it foreign_vat_country_id, you're implying it's an id, so an integer. Here, it's a record; this causes confusion => foreign_vat_country